### PR TITLE
chore(`anvil`): Add base alloy types

### PR DESCRIPTION
**Pull request by Evalir**
This PR adds an alloy version of the types in `anvil-core`, replacing them with `alloy-consensus` types. We keep our own `TypedTransactionRequest`/`TypedTransaction`/`TypedReceipt` types to enable optimism support.

Note that this PR only adds types. It doesn't remove or migrate the old types as to not blow up the scope. Incremental PRs removing old code will follow.

### CODE REVIEW

File: crates/anvil/core/src/eth/transaction/alloy.rs [hunk 1]
```diff
@@ -0,0 +1,379 @@
+use alloy_consensus::{TxEip2930, TxEip1559, TxLegacy};
+use alloy_network::{TxKind, Signed};
+use alloy_primitives::{Address, B256, Bloom, Bytes, Log, U256};
+use alloy_rlp::{Encodable, Decodable, Header as RlpHeader, Error as DecodeError};
+use alloy_rpc_types::AccessList;
```
Comment (gakonst) [inline crates/anvil/core/src/eth/transaction/alloy.rs:5]: @DaniPopes we prob want to figure out a nice re-export for these

File: crates/anvil/core/src/eth/transaction/alloy.rs [hunk 2]
```diff
@@ -0,0 +1,1243 @@
...
+                    },
+                    ..Default::default()
+                }
+            }
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum TypedTransaction {
```
Comment (mattsse) [inline crates/anvil/core/src/eth/transaction/alloy.rs:518]: docs

File: crates/anvil/core/src/eth/transaction/alloy.rs [hunk 3]
```diff
@@ -0,0 +1,1243 @@
...
+            Ordering::Equal => {
+                Err(alloy_rlp::Error::Custom("an empty list is not a valid transaction encoding"))
+            }
+            Ordering::Greater => {
+                <Signed<TxLegacy> as Decodable>::decode(buf).map(TypedTransaction::Legacy)
+            }
+        }
+    }
+}
+
```
Comment (mattsse) [inline crates/anvil/core/src/eth/transaction/alloy.rs:803]: docs

File: crates/anvil/core/src/eth/utils.rs [hunk 4]
```diff
@@ -33,3 +34,9 @@ pub fn alloy_to_revm_access_list(list: Vec<AlloyAccessListItem>) -> Vec<(Address
         .map(|item| (item.address, item.storage_keys.into_iter().map(|k| k.into()).collect()))
         .collect()
 }
+
+pub fn eip_to_revm_access_list(list: Vec<AlloyEipAccessListItem>) -> Vec<(Address, Vec<U256>)> {
```
Comment (mattsse) [inline crates/anvil/core/src/eth/utils.rs:38]: docs

File: crates/anvil/core/src/eth/transaction/alloy.rs [hunk 5]
```diff
@@ -0,0 +1,1243 @@
...
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum TypedTransactionRequest {
+    Legacy(TxLegacy),
+    EIP2930(TxEip2930),
+    EIP1559(TxEip1559),
+    Deposit(DepositTransactionRequest),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct DepositTransactionRequest {
```
Comment (mattsse) [inline crates/anvil/core/src/eth/transaction/alloy.rs:207]: let's move all deposit stuff into a separate file, this way we can move this to alloy more easily.

File: crates/anvil/core/src/eth/transaction/alloy.rs [hunk 6]
```diff
@@ -0,0 +1,1243 @@
+use std::ops::Deref;
+
```
Comment (mattsse) [inline crates/anvil/core/src/eth/transaction/alloy.rs:2]:
```suggestion
```

**Review by mattsse — APPROVED** [review]
nits, otherwise lgtm

**Pull request merged**